daft/context.py

```diff
@@ -178,13 +178,49 @@ def is_ray_runner(self) -> bool:
             return isinstance(runner_config, _RayRunnerConfig)
 
 
+@dataclasses.dataclass
+class DaftActorContext:
+    """Actor-local context for the current Daft stateful UDF environment"""
+
+    _running_in_actor: bool = False
+    _rank: int | None = None
+    _resource_request: ResourceRequest | None = None
+
+    @property
+    def rank(self) -> int | None:
+        """Rank of the current actor"""
+        return self._rank
+
+    @property
+    def resource_request(self) -> ResourceRequest | None:
+        """Resource request of the current actor"""
+        return self._resource_request
+
+
 _DaftContext = DaftContext()
 
 
 def get_context() -> DaftContext:
     return _DaftContext
 
 
+_DaftActorContext = DaftActorContext()
```

Comment:
This seems to not be used at all except for in testing? We should get rid of it if that's the case, very confusing.

Reply:
It's meant to be used by a stateful UDF when it calls `daft.contextget_actor_context()` to get information such as its rank